src/external/tsa.ts

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2023 The Sigstore Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import fetch, { FetchInterface } from 'make-fetch-happen';
+import { ua } from '../util';
+import { checkStatus } from './error';
+
+export interface TimestampRequest {
+  artifactHash: string;
+  hashAlgorithm: string;
+  certificates?: boolean;
+  nonce?: number;
+  tsaPolicyOID?: string;
+}
+
+export interface TimestampAuthorityOptions {
+  baseURL: string;
+}
+
+export class TimestampAuthority {
+  private fetch: FetchInterface;
+  private baseUrl: string;
+
+  constructor(options: TimestampAuthorityOptions) {
+    this.fetch = fetch.defaults({
+      retry: { retries: 2 },
+      timeout: 5000,
+      headers: {
+        'Content-Type': 'application/json',
+        'User-Agent': ua.getUserAgent(),
+      },
+    });
+    this.baseUrl = options.baseURL;
+  }
+
+  public async createTimestamp(request: TimestampRequest): Promise<Buffer> {
+    const url = `${this.baseUrl}/api/v1/timestamp`;
```

Comment:
Is this in a spec somewhere or how do we know it's always `/api/v1/timestamp`?

Reply:
Seems like a sigstore thing from googling around: https://github.com/sigstore/timestamp-authority

Seems fine, but guess this means any sigstore client only supports the "sigstore" flavoured TSA setup.

Reply:
Yeah, this endpoint and the request format we're using is Sigstore-specific